src/cryptography/hazmat/primitives/padding.py

```diff
@@ -122,3 +128,26 @@ def finalize(self):
         res = self._buffer[:-pad_size]
         self._buffer = None
         return res
+
+
+class ANSIX923(PKCS7):
+
+    def padder(self):
+        return _ANSIX923PaddingContext(self.block_size)
+
+    def unpadder(self):
+        return _ANSIX923UnpaddingContext(self.block_size)
+
+
+@utils.register_interface(PaddingContext)
+class _ANSIX923PaddingContext(_PKCS7PaddingContext):
+
+    def _padding(self, size):
+        return six.int2byte(0) * (size - 1) + six.int2byte(size)
+
+
+@utils.register_interface(PaddingContext)
+class _ANSIX923UnpaddingContext(_PKCS7UnpaddingContext):
+
+    def _check_padding(self):
+        return True
```

Comment:
Shouldn't we check that all the bytes are 0? Without that you are
vulnerable to attacks such as the one on SSLv3.

On Fri, Feb 26, 2016 at 5:14 PM, Cédric Krier <notifications@github.com>
wrote:

> In src/cryptography/hazmat/primitives/padding.py
> <https://github.com/pyca/cryptography/pull/2736#discussion_r54309644>:
>
> > +    def unpadder(self):
> > +        return _ANSIX923UnpaddingContext(self.block_size)
> > +
> > +
> > +@utils.register_interface(PaddingContext)
> > +class _ANSIX923PaddingContext(_PKCS7PaddingContext):
> > +
> > +    def _padding(self, size):
> > +        return six.int2byte(0) * (size - 1) + six.int2byte(size)
> > +
> > +
> > +@utils.register_interface(PaddingContext)
> > +class _ANSIX923UnpaddingContext(_PKCS7UnpaddingContext):
> > +
> > +    def _check_padding(self):
> > +        return True
>
> Why? There is no check in ANSI X.923 padding.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2736/files#r54309644>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084